frame/nomination-pools/src/lib.rs

```diff
@@ -1258,34 +1381,32 @@ pub mod pallet {
 		pub fn unbond_other(
 			origin: OriginFor<T>,
 			delegator_account: T::AccountId,
+			mut unbonding_points: BalanceOf<T>,
 		) -> DispatchResult {
 			let caller = ensure_signed(origin)?;
 			let (mut delegator, mut bonded_pool, mut reward_pool) =
 				Self::get_delegator_with_pools(&delegator_account)?;
-
 			bonded_pool.ok_to_unbond_other_with(&caller, &delegator_account, &delegator)?;
```

Comment:
We need to pass in `unbonding_points` to `ok_to_unbond_other_with` and if delegator account is the depositor we need to check that `unbonding_points == delegator.active_points` because we can't allow the depositor to partially unbond. We should also make sure in the tests that the depositor cannot partially unbond. Partial unbonding for the depositor destroys the guarantee that the depositor has the expected skin in the game

Reply:
I just want to make sure I understand why: Does some code semantic break if we allow this? because it nothing seems broken right now. Or do you mean that the logical assumption that the depositor's self-stake should be constant is broken?

Reply:
It seems to me like we want to enforce each depositor to have at least `MinCreateBond` skin in the game, right? in this case, in this case I agree. And can't we just make sure the depositor cannot call `unbond_other` at all, regardless of `unbonding_points`, until the pool is both empty and in `Destorying` state?

Reply:
also, what if the delegator does `bond_extra`? 

All in all the assumption that we're trying to maintain is not super clear to me. Let's first make sure we're on the same page about that, then doing it will be the easier part.

Reply:
As discussed on call, we just want to make sure the depositor always has MinCreateBond, anything above we will let them unbond and withdraw. This check should happen in unbond